django/db/models/expressions.py

```diff
@@ -834,11 +833,11 @@ def copy(self):
         c.cases = c.cases[:]
         return c
 
-    def as_sql(self, compiler, connection, template=None, extra=None):
+    def as_sql(self, compiler, connection, **kwargs):
         connection.ops.check_expression_support(self)
         if not self.cases:
             return compiler.compile(self.default)
-        template_params = dict(extra) if extra else {}
+        template_params = kwargs.get('extra', {})
```

Comment:
This should just be `template_params = kwargs`. The `extra` in the signature that was removed is acting like a `kwargs` anyway.